src/webui/api/searchcontroller.cpp

```diff
@@ -0,0 +1,286 @@
+
+    const QStringList names = params()["names"].split('|');
+    for (const QString &name : names)
+        SearchPluginManager::instance()->uninstallPlugin(name.trimmed());
+}
+
+void SearchController::enablePluginAction()
+{
+    checkParams({"names", "enable"});
+
+    const QStringList names = params()["names"].split('|');
+    const bool enable = parseBool(params()["enable"].trimmed(), false);
+
+    for (const QString &name : names)
+        SearchPluginManager::instance()->enablePlugin(name.trimmed(), enable);
+}
+
+void SearchController::updatePluginsAction()
+{
+    SearchPluginManager *pluginManager = SearchPluginManager::instance();
+
+    connect(pluginManager, &SearchPluginManager::checkForUpdatesFinished, this, &SearchController::checkForUpdatesFinished);
+    // connect(pluginManager, &SearchPluginManager::checkForUpdatesFailed, this, &SearchController::checkForUpdatesFailed);
+    pluginManager->checkForUpdates();
+}
+
+void SearchController::checkForUpdatesFinished(const QHash<QString, PluginVersion> &updateInfo)
+{
+    if (updateInfo.isEmpty()) {
+        qDebug() << "All your plugins are already up to date.";
+        return;
+    }
+
+    SearchPluginManager *pluginManager = SearchPluginManager::instance();
+    foreach (const QString &pluginName, updateInfo.keys()) {
+        qDebug() << "Updating plugin" << pluginName;
+        pluginManager->updatePlugin(pluginName);
+    }
+}
+
+QString SearchController::generateSearchId() const
+{
+    QString id;
+
+    do {
+        const size_t size = 2;
+        quint32 tmp[size];
+
+        for (size_t i = 0; i < size; ++i)
+            tmp[i] = Utils::Random::rand();
+
+        id = QByteArray::fromRawData(reinterpret_cast<const char *>(tmp), sizeof(quint32) * size).toBase64();
+    }
+    while (m_searches.contains(id));
+
+    return id;
+}
+
+/**
+ * Returns the search results in JSON format.
+ *
+ * The return value is an object with a status and an array of dictionaries.
+ * The dictionary keys are:
+ *   - "fileName"
+ *   - "fileUrl"
+ *   - "fileSize"
+ *   - "nbSeeders"
+ *   - "nbLeechers"
+ *   - "siteUrl"
+ *   - "descrLink"
+ */
+QMap<QString, QVariant> SearchController::getResults(const QList<SearchResult> searchResults, const bool isSearchActive, const int queueSize) const
+{
+    QVariantList searchResultsVariantList;
+    for (const SearchResult searchResult : searchResults) {
+        QMap<QString, QVariant> searchResultMap;
+
+        searchResultMap.insert("fileName", searchResult.fileName);
+        searchResultMap.insert("fileUrl", searchResult.fileUrl);
+        searchResultMap.insert("fileSize", searchResult.fileSize);
+        searchResultMap.insert("nbSeeders", searchResult.nbSeeders);
+        searchResultMap.insert("nbLeechers", searchResult.nbLeechers);
+        searchResultMap.insert("siteUrl", searchResult.siteUrl);
+        searchResultMap.insert("descrLink", searchResult.descrLink);
+
+        searchResultsVariantList << searchResultMap;
```

Comment:
```c++
searchResultsVariantList += {
    { "fileName", searchResult.fileName }
    //...
};
```

Reply:
I get this error when compiling with `+=` and `<<`

`
webui/api/searchcontroller.cpp:233:34: error: no match for ‘operator+=’ (operand types are ‘QVariantList {aka QList<QVariant>}’ and ‘<brace-enclosed initializer list>’)
`

Reply:
> I get this error when compiling with += and <<

pardon, I updated the code above.

Reply:
>pardon, I updated the code above.

As I requested before, it should be QJsonObject/QJsonArray here too.